elastic/kibana
# [Upgrade Assistant] Remove ML/Watcher logic

### Summary

Currently UA shows in the ES deprecations page deprecations for system indices, and since for 7.16 system indices will be resolved separately through the system indices migration flow that will be shown in the Overview page (see https://github.com/elastic/kibana/pull/110593) we will have to remove them from the deprecations table (see https://github.com/elastic/kibana/issues/113280). At the moment since they are shown to the user and we allow them to resolve these deprecations, there were a few considerations we had to have when dealing with the reindexing of ML and Watcher indices:

* **ML**: If we would detect that a reindexing operation is performed on one of the following system indices `'.ml-state', '.ml-anomalies', '.ml-config'` then we would attempt to go ahead and set ML upgrade mode flag depending on whether we are starting or stopping the job. This was needed because upgrading ML indices requires setting the mlUpgradeMode setting  accordingly, otherwise the re-indexing process would fail.
* **Watcher**:  If we would detect that a reindexing operation is performed on one of the following system indices `'.watches', '.triggered-watches'`  we would start/stop the watcher accordingly depending on whether we are stopping or starting the job. Doing so was crucial because those indices cannot be re-indexed if the watcher is started.

 In terms of UX, this had a minimal impact for user since the only difference between a regular deprecation that required re-indexing and one for any of the above mentioned indices was just 2 extra steps for pausing ml_jobs/watcher before doing anything and another one for resuming the ml_job/watcher afterwards:

<img width="307" alt="Screenshot 2021-09-30 at 14 20 40" src="https://user-images.githubusercontent.com/1191206/135453848-7f1c61be-8312-42c2-9f95-8dca727e13bf.png">


Therefore this PR removes the logic that was already setup for:
* dealing with stopping/resuming ML jobs when reindexing ML system indices 
* dealing with stopping/resuming watcher when reindexing watcher system indices
* showing extra steps for the above cases in progress component shown in the reindexing flyout

**How to test**

1. Unzip [6.8-data-snapshot.zip](https://github.com/elastic/kibana/files/7242690/6.8-data-snapshot.zip) and start up ES as follows: `yarn es snapshot -E path.data=/path/to/6.8-data-snapshot`. This contains indices and ML job model snapshots created on 6.8. This will trigger the reindex and ML deprecations.
2. Start up Kibana, `yarn start`.
3. Navigate to UA, open `ES deprecation issues`
4. Verify that for reindex deprecations for the above mentioned system indices there are no extra steps regarding pausing/resuming ml jobs or watcher.

If after following the steps you don't see any deprecations for es in upgrade assistant, try unzipping the folder in the `kibana/.es` folder and starting up es with: `yarn es snapshot -E path.data=../6.8-data-snapshot`

Note: We should hide system indices in the deprecations list after the ES team finishes the work for migrating system indices (see https://github.com/elastic/kibana/issues/113280)
